### PR TITLE
Adding login feature for users to leave comments

### PR DESCRIPTION
**Pull request by shreyabarua**
Provides functionality for users to first log in before they leave a comment

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
     <script src="script.js"></script>
   </head>
-  <body onload="getMessages()">
+  <body onload="getMessages(); displayComments()">
```
Comment (astrelni) [inline portfolio/src/main/webapp/index.html]: Consider adding a separate function named something like `onLoad` that calls `getMessages` and `displayComments`.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -61,3 +61,16 @@ function createListElement(text) {
...
 }
+
+function displayComments() {
+    fetch('/login').then(response => response.json()).then((login) => {
+        if (login.status) {
+            document.getElementById("comments-form").style.display = 'block';
+        }
+        else {
+            const loginLinkElement = document.getElementById('login-link-container');
+            loginLinkElement.innerHTML="<p>Login <a href=\"" + login.login_url + "\">here</a> to leave a comment.</p>";
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js]: The outer `<p></p>` seems unnecessary, `#login-link-container` is already a `<p>`.

Nit: I think you want "Log in" for verb as opposed to "Login" the noun.

**Review by astrelni — DISMISSED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    UserService userService = UserServiceFactory.getUserService();
+    boolean status = userService.isUserLoggedIn();
+    String json_login;
```
Comment (oydang) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Java style guide: variables should be named with camel case. `jsonLogin`. https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -166,18 +166,21 @@ <h2 id = "contact" > Contact </h2>
...
+        <br/><br/>
+        <input type="submit" />
+        </form>
+        
+        <p>Comments:</p>
+        <ul id="msg-container"></ul>
+        <button onclick="deleteComments()">delete comments</button>
+    </div>
+    <p id="login-link-container"></p>
+    <!--<button onclick="displayComments()">display</button>-->
```
Comment (oydang) [inline portfolio/src/main/webapp/index.html]: Delete this commented out line?

**Review by oydang — CHANGES_REQUESTED** [review]

**Review by oydang — APPROVED** [review]

**Pull request merged**